pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -389,3 +390,29 @@ func (p *Provider) getInstanceProfile(nodeClass *v1beta1.EC2NodeClass) (string,
 	}
 	return "", errors.New("neither spec.instanceProfile or spec.role is specified")
 }
+
+func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) error {
+	clusterName := options.FromContext(ctx).ClusterName
+	var ltNames []*string
+	if err := p.ec2api.DescribeLaunchTemplatesPagesWithContext(ctx, &ec2.DescribeLaunchTemplatesInput{
+		Filters: []*ec2.Filter{
+			{Name: aws.String(fmt.Sprintf("tag:%s", karpenterManagedTagKey)), Values: []*string{aws.String(clusterName)}},
+			{Name: aws.String(fmt.Sprintf("tag:%s", v1beta1.LabelNodeClass)), Values: []*string{aws.String(nodeClass.Name)}},
+		},
+	}, func(output *ec2.DescribeLaunchTemplatesOutput, _ bool) bool {
+		for _, lt := range output.LaunchTemplates {
+			ltNames = append(ltNames, lt.LaunchTemplateName)
+		}
+		return true
+	}); err != nil {
+		return fmt.Errorf("fetching launch templates, %w", err)
+	}
+
+	for _, name := range ltNames {
```

Comment:
I wonder if we should best-effort try to delete the launch templates and then return a multi-err here that way we delete as many launch templates as possible here.